src/main/kotlin/org/rust/lang/core/resolve/NameResolution.kt

```diff
@@ -248,6 +265,10 @@ fun processPathResolveVariants(lookup: ImplLookup, path: RsPath, isCompletion: B
         else -> TYPES_N_VALUES
     }
 
+    if (isCompletion && parent is RsPathExpr && qualifier?.path == null) {
```

Comment:
Let's add a comment with an explanation that without `!` we parse `crate::path_prefix` as `RsPathExpr`